packages/react-dom/src/__tests__/ReactDOMServerIntegrationElements-test.js

```diff
@@ -871,5 +871,72 @@ describe('ReactDOMServerIntegration', () => {
             : ''),
       );
     });
+
+    describe('badly-typed elements', function() {
+      beforeEach(() => {
+        if (__DEV__) {
+          spyOnDev(console, 'error');
```

Comment:
Why are we doing this here instead of wrapping `<UndefinedComponent />` JSX call where it happens?

```
let el;
expect(() => {
  el = <UndefinedComponent />;
}).toWarnInDev(...)
await render(el);
```

Reply:
Didn't make the connection that I could do this. Thanks, this is so much easier than overriding the function. Removed the `beforeEach` and added this to the tests